waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -130,6 +139,361 @@ protected String getUsage() {
+                            }
+
+                            PartitionValidationResults partitionValidationResults = partitionValidationResultsList.get(partitionId);
+                            ValidationResults validationResults = new ValidationResults(ValidationResults.ValidationType.PARTITION_ASSIGNMENT_ZK_SERVER_CONSISTENCY,
+                                                                        status, error);
+                            partitionValidationResults.validationResultsMap.put(validationResults.type, validationResults);
+                        });
+                    }).exceptionally(e -> {
+                        for (PartitionInfo partitionInfo : zookeeperAssignments) {
+                            PartitionValidationResults partitionValidationResults = partitionValidationResultsList.get(partitionInfo.partitionId);
+                            ValidationResults validationResults = new ValidationResults(
+                                    ValidationResults.ValidationType.PARTITION_ASSIGNMENT_ZK_SERVER_CONSISTENCY,
+                                    ValidationResults.Status.FAILURE,
+                                    e.getMessage());
+                            partitionValidationResults.validationResultsMap.put(validationResults.type, validationResults);
+                        }
+                        return null;
+                    });
+        }
+
+        /**
+         *
+         * Validate for all servers in the cluster the consistency of partition assignments on the actual servers
+         * versus on Zookeeper metadata.
+         *
+         * @param rpcClient Client used to connect to Waltz servers to fetch partition assignments
+         * @param clusterManager Contains zkclient used to fetch partition assignments of Zookeeper
+         * @param partitionValidationResultsList List indexed by partition in which validation outputs are inserted
+         * @return Completable future that complete once the partitionValidationResultsList is filled with validation data
+         * from all partitions
+         * @throws ClusterManagerException Thrown if Zookeeper is missing some ZNodes or ZNode values
+         * @throws InterruptedException If thread interrupted while waiting for channel with Waltz servers to be ready
+         */
+        private CompletableFuture<Void> buildServersZKPartitionAssignmentsConsistencyValidation(InternalRpcClient rpcClient,
+                                                                                                ClusterManager clusterManager,
+                                                                                                List<PartitionValidationResults> partitionValidationResultsList)
+                                                                    throws ClusterManagerException, InterruptedException {
+            Set<CompletableFuture> set = new HashSet<>();
```

Comment:
nit: can you rename this to something else. Isn't intuitive to understand what it does.

Reply:
fixed. Thanks!